storage/remote/otlptranslator/prometheusremotewrite/metrics_to_prw.go

```diff
@@ -91,8 +92,10 @@ func (c *PrometheusConverter) FromMetrics(ctx context.Context, md pmetric.Metric
 
 				metric := metricSlice.At(k)
 				mostRecentTimestamp = max(mostRecentTimestamp, mostRecentTimestampInMetric(metric))
-
-				if !isValidAggregationTemporality(metric) {
+				temporality, hasTemporality := aggregationTemporality(metric)
+				if hasTemporality &&
+					(temporality == pmetric.AggregationTemporalityUnspecified ||
+						(!settings.AllowDeltaTemporality && temporality == pmetric.AggregationTemporalityDelta)) {
```

Comment:
I'm not understanding very well what we want to do with Unspecified temporality. We're starting to refuse them only now? If we accepted before, shouldn't we keep the behavior?

Reply:
AggregationTemporalityUnspecified has always resulted in an error for any types that have temporality, this behaviour is being preserved.

Previously the only valid temporality (for types that have temporality) was cumulative:
https://github.com/prometheus/prometheus/blob/f141e8ad734ccc85df66d5a081ca58b348ed43dd/storage/remote/otlptranslator/prometheusremotewrite/helper.go#L229-L242 (so implicitly unspecified temporality was invalid).

Reply:
I've rewritten this part to focus on the valid temporalities 

https://github.com/prometheus/prometheus/blob/d5c4cdf9c2c4ee5386088953fcc6e744141cb9a0/storage/remote/otlptranslator/prometheusremotewrite/metrics_to_prw.go#L101-L109

Does that make it clearer?